main.py

```diff
@@ -7,37 +7,45 @@
 
 from messaging import Messenger
 from process import Process
-
-
-# Configure logging
-logging.basicConfig(format='%(asctime)s %(name)s: %(message)s',
-    filename='NRLog', level=logging.DEBUG)
-logger = logging.getLogger('Core')
+import nr_logger
 
 
 def main():
-  logger.info('System starting')
   messenger = Messenger()
-  logger.debug('Messenger finished initializing')
 
-  processes = {}
+  # This has to happen before anyone uses a logger.
+  nr_logger.QueueLogger.set_queue(messenger.get_queue("logging"))
+
+  # Configure our logger.
+  logger = logging.getLogger('Core')
+  logger.info('System starting...')
+
+  processes = []
+  processes.append(nr_logger.LogWriter(messenger, "NRLog"))
 
   #TODO: spawn BeliefSystem, RadioListener, MAVLink
   #TODO: start processes
 
+  # Start all the processes.
+  logger.info("Starting processes.")
+  pids = {}
+  for process in processes:
+    pid = process.start()
+    pids[pid] = process
+
   # Restart any processes that die.
   while True:
     # Wait for any child process.
     pid, status = os.waitpid(-1, 0)
     if status != 0:
       logger.warning("Process %d exited unexpectedly! Restarting..." % (pid))
-      processes[pid].start()
+      new_pid = pids[pid].start()
+      pids[new_pid] = pids[pid]
+      del pids[pid]
 
 
-if __name__ == "__main__":
-  try:
-    main()
-  except Exception as e:
-    logger.critical("Core: %s" % (e))
-
```

Comment:
Maybe make a static emergency logger that just dumps to a log file that includes class name, PID, and epoch time